webodf/tests/gui/ZoomHelperTests.js

```diff
@@ -0,0 +1,226 @@
+ * @implements {core.UnitTest}
+ */
+gui.ZoomHelperTests = function ZoomHelperTests(runner) {
+    "use strict";
+    var r = runner,
+        t;
+
+    this.setUp = function () {
+        var testArea = core.UnitTest.provideTestAreaDiv();
+        t = {
+            zoomHelper : new gui.ZoomHelper(),
+            scrollPane : testArea,
+            document : testArea.ownerDocument
+        };
+        t.scrollPane.style.position = "relative";
+    };
+
+    this.tearDown = function () {
+        t = {};
+        core.UnitTest.cleanupTestAreaDiv();
+    };
+
+    function setupScrollPane(width, height) {
+        t.scrollPane.style.width = width + "px";
+        t.scrollPane.style.height = height + "px";
+        t.scrollPane.style.overflow = "auto";
+    }
+
+    function setupInnerScrollPane(width, height) {
+        t.innerScrollPane = t.document.createElement("div");
+        t.innerScrollPane.id = "innerScrollPane";
+        t.innerScrollPane.style.transformOrigin = "top left";
+        t.innerScrollPane.style.width = width + "px";
+        t.innerScrollPane.style.height = height + "px";
+        t.innerScrollPane.style.position = "relative";
+        t.innerScrollPane.style.overflow = "auto";
+        t.innerScrollPane.style.border = "1px solid green";
+        t.scrollPane.appendChild(t.innerScrollPane);
+    }
+
+    function setupSimulatedContent(width, height) {
+        t.simulatedContent = t.document.createElement("div");
+        t.simulatedContent.id = "simulatedContent";
+        t.simulatedContent.style.transformOrigin = "top left";
+        t.simulatedContent.style.width = width + "px";
+        t.simulatedContent.style.height = height + "px";
+        t.simulatedContent.style.position = "relative";
+        t.simulatedContent.style.overflow = "hidden";
+        t.simulatedContent.style.border = "1px solid black";
+        for(var i = 0; i < 1000; i++) {
+            t.simulatedContent.appendChild(t.document.createTextNode(i + "_" + i + " "));
+        }
+    }
+
+    function setupRedSquare(top, left) {
+        var redSquare = t.document.createElement("div");
+        redSquare.id = "redSquare";
+        redSquare.style.position = "absolute";
+        redSquare.style.width = "5px";
+        redSquare.style.height = "5px";
+        redSquare.style.top = top + "px";
+        redSquare.style.left = left + "px";
+        redSquare.style.backgroundColor = "red";
+        t.simulatedContent.appendChild(redSquare);
+    }
+
+    function setZoomLevel_Maintains_scroll_position_on_zoom_in() {
+        setupScrollPane(100, 100);
+        setupSimulatedContent(200, 200);
+        t.scrollPane.appendChild(t.simulatedContent);
+        t.zoomHelper.setZoomableElement(t.simulatedContent);
+        setupRedSquare(50, 80);
+
+        t.scrollPane.scrollTop = 50;
+        t.scrollPane.scrollLeft = 80;
+
+        var zoom = 2;
+        t.zoomHelper.setZoomLevel(zoom);
+
+        r.shouldBe(t, "t.scrollPane.scrollTop", "100", 2);
```

Comment:
I'd suggest a class const for `2` in this test file, as it seems on initial glance like this is somehow related to the zoom levels.